src/components/ActionDetail/ActionDetail.test.tsx

```diff
@@ -0,0 +1,161 @@
+    });
+
+    it('has selector value in input', () => {
+      const { getByLabelText } = render(
+        <ActionDetail actionContext={clickAction} actionIndex={0} stepIndex={0} />
+      );
+      const input = getByLabelText(/click/i);
+      expect(input).toBeTruthy();
+      expect(input.getAttribute('value')).toEqual('#elem-id');
+    });
+
+    it('updates selector', () => {
+      const onUpdateActionMock = jest.fn(updatedAction => {
+        return updatedAction;
+      });
+      const { getByLabelText, getByRole } = renderWithStepsContext(
+        <ActionDetail actionContext={clickAction} actionIndex={0} stepIndex={0} />,
+        { onUpdateAction: onUpdateActionMock }
+      );
+      const input = getByLabelText(/click/i);
+
+      fireEvent.change(input, { target: { value: '#new-elem-id' } });
+      const button = getByRole('button', { name: /save/i });
+      fireEvent.click(button);
+      expect(onUpdateActionMock).toHaveBeenCalled();
+      const {
+        value: { action },
+      } = onUpdateActionMock.mock.results[0];
+      expect(action.selector).toEqual('#new-elem-id');
+    });
+  });
+
+  describe('fill', () => {
+    let fillAction: ActionContext;
+    beforeEach(() => {
+      fillAction = createAction('fill', {
+        action: { selector: '#elem-id', url: '', text: 'Hello world' },
+      });
+    });
+
+    it('has two input fields', () => {
+      const { getByLabelText } = render(
+        <ActionDetail actionContext={fillAction} actionIndex={0} stepIndex={0} />
+      );
+      const input1 = getByLabelText(/fill/i);
+      expect(input1).toBeTruthy();
+      expect(input1.getAttribute('value')).toEqual('#elem-id');
+
+      const input2 = getByLabelText(/Value/i);
+      expect(input2).toBeTruthy();
+      expect(input2.getAttribute('value')).toEqual('Hello world');
+    });
+
+    it('updates selector and fill value', () => {
+      const onUpdateActionMock = jest.fn(updatedAction => {
+        return updatedAction;
+      });
+      const { getByLabelText, getByRole } = renderWithStepsContext(
+        <ActionDetail actionContext={fillAction} actionIndex={0} stepIndex={0} />,
+        { onUpdateAction: onUpdateActionMock }
+      );
+      const selectorInput = getByLabelText(/fill/i);
+      fireEvent.change(selectorInput, { target: { value: '#new-elem-id' } });
+
+      const valueInput = getByLabelText(/value/i);
+      fireEvent.change(valueInput, { target: { value: 'Elastic' } });
+      const button = getByRole('button', { name: /save/i });
+      fireEvent.click(button);
+      expect(onUpdateActionMock).toHaveBeenCalled();
+      const {
+        value: { action },
+      } = onUpdateActionMock.mock.results[0];
+      expect(action.selector).toEqual('#new-elem-id');
+      expect(action.text).toEqual('Elastic');
+    });
+  });
+});
+
+function renderWithStepsContext(childComp: React.ReactChild, overrides?: Partial<IStepsContext>) {
```

Comment:
I think we don't need this, have a look at [this helper function](https://github.com/elastic/synthetics-recorder/blob/acd37f14df07399cfdd4a96b2bb6694f5796c9db/src/helpers/test/render.tsx#L40).

Reply:
Oh nice!

Reply:
I've updated it using the helper function 👍